src/cljs/bluegenes/route.cljs

```diff
@@ -0,0 +1,230 @@
+;; between `:set-active-panel` and the router events.
+(def routes
+  ["/"
+   [":mine"
+    {:controllers
+     [{:parameters {:path [:mine]}
+       :start (fn [{{:keys [mine]} :path}]
+                (dispatch [:init-mine mine]))}]}
+    [""
+     {:name ::home
+      :controllers
+      [{:start #(dispatch [:set-active-panel :home-panel])}]}]
+    ["/debug/:panel"
+     {:name ::debug
+      :controllers
+      [{:parameters {:path [:panel]}
+        :start (fn [{{:keys [panel]} :path}]
+                 (dispatch [:set-active-panel :debug-panel
+                            nil
+                            [:bluegenes.pages.developer.events/panel panel]]))}]}]
+    ["/help"
+     {:name ::help
+      :controllers
+      [{:start #(dispatch [:set-active-panel :help-panel])}]}]
+    ["/templates"
+     {:name ::templates
+      :controllers
+      [{:start #(dispatch [:set-active-panel :templates-panel])}]}]
+    ["/upload"
+     [""
+      {:name ::upload
+       :controllers
+       [{:start #(dispatch [:set-active-panel :upload-panel])}]}]
+     ["/:step"
+      {:name ::upload-step
+       :controllers
+       [{:parameters {:path [:step]}
+         :start (fn [{{:keys [step]} :path}]
+                  (dispatch [:set-active-panel :upload-panel
+                             {:step (keyword step)}]))}]}]]
+    ["/explore"
+     {:name ::explore
+      :controllers
+      [{:start #(dispatch [:set-active-panel :explore-panel])}]}]
+    ["/search"
+     {:name ::search
+      :controllers
+      [{:parameters {:query [:keyword]}
+        :start (fn [{{:keys [keyword]} :query}]
+                 (dispatch [:set-active-panel :search-panel
+                            nil
+                            [:search/full-search keyword]]))}]}]
+    ["/querybuilder"
+     {:name ::querybuilder
+      :controllers
+      [{:start #(dispatch [:set-active-panel :querybuilder-panel
+                           nil
+                           [:qb/make-tree]])}]}]
+    ["/list/:title"
+     {:name ::list
+      :controllers
+      [{:parameters {:path [:title]}
+        :start (fn [{{:keys [title]} :path}]
+                 (dispatch [:set-active-panel :results-panel
+                            nil
+                            [::view-list title]]))}]}]
+    ["/regions"
+     {:name ::regions
+      :controllers
+      [{:start #(dispatch [:set-active-panel :regions-panel])}]}]
+    ["/mymine"
+     {:name ::mymine
+      :controllers
+      [{:start #(dispatch [:set-active-panel :mymine-panel])}]}]
+    ["/report/:type/:id"
+     {:name ::report
+      :controllers
+      [{:parameters {:path [:mine :type :id]}
+        :start (fn [{{:keys [mine type id]} :path}]
+                 (dispatch [:set-active-panel :reportpage-panel
+                            {:type type :id id :mine mine}
+                            [:load-report mine type id]]))}]}]]])
+;; You can do initialisations by adding a :start function to :controllers.
```

Comment:
I'm not quite sure what you mean here by teardowns and initialisations here, or why I might want to use them.

Reply:
Most of the routes have a `:controllers` vector with a `:start` map. While you can copy-cat the existing `:start` maps when you create new routes, there aren't any using `:stop` (teardowns) currently. I wanted to make sure to write it somewhere so people know they exist as well.